File: src/main/java/com/mbayou/kick4k/channels/Channel.java
package com.mbayou.kick4k.channels;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.mbayou.kick4k.categories.Category;

public class Channel {
    private final String bannerPicture;
    private final Integer broadcasterUserId;
    private final Category category;
    private final String channelDescription;
    private final String slug;
    private final StreamInformation stream;
    private final String streamTitle;

    @JsonCreator
    public Channel(@JsonProperty("banner_picture") String bannerPicture,
                   @JsonProperty("broadcaster_user_id") Integer broadcasterUserId,
                   @JsonProperty("category") Category category,
                   @JsonProperty("channel_description") String channelDescription,
                   @JsonProperty("slug") String slug,
                   @JsonProperty("stream") StreamInformation stream,
                   @JsonProperty("stream_title") String streamTitle) {
        this.bannerPicture = bannerPicture;
        this.broadcasterUserId = broadcasterUserId;
        this.category = category;
        this.channelDescription = channelDescription;
        this.slug = slug;
        this.stream = stream;
        this.streamTitle = streamTitle;
    }

    public String getBannerPicture() {
        return bannerPicture;
    }

    public Integer getBroadcasterUserId() {
        return broadcasterUserId;
    }

    public Category getCategory() {
        return category;
    }

    public String getChannelDescription() {
        return channelDescription;
    }

    public String getSlug() {
        return slug;
    }

    public StreamInformation getStream() {
        return stream;
    }

    public String getStreamTitle() {
        return streamTitle;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/channels/Channel.kt
package com.mbayou.kick4k.channels

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.mbayou.kick4k.categories.Category

data class Channel @JsonCreator constructor(
    @JsonProperty("banner_picture") val bannerPicture: String?,
    @JsonProperty("broadcaster_user_id") val broadcasterUserId: Int?,
    @JsonProperty("category") val category: Category?,
    @JsonProperty("channel_description") val channelDescription: String?,
    @JsonProperty("slug") val slug: String?,
    @JsonProperty("stream") val stream: StreamInformation?,
    @JsonProperty("stream_title") val streamTitle: String?,
)
